app/controllers/api/users.php

```diff
@@ -781,18 +792,39 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
     ->param('userId', '', new UID(), 'User ID.')
     ->param('password', '', new Password(), 'New user password. Must be at least 8 chars.')
     ->inject('response')
+    ->inject('project')
     ->inject('dbForProject')
     ->inject('events')
-    ->action(function (string $userId, string $password, Response $response, Database $dbForProject, Event $events) {
+    ->action(function (string $userId, string $password, Response $response, Document $project, Database $dbForProject, Event $events) {
 
         $user = $dbForProject->getDocument('users', $userId);
 
         if ($user->isEmpty()) {
             throw new Exception(Exception::USER_NOT_FOUND);
         }
 
+        $newPassword = Auth::passwordHash($password, Auth::DEFAULT_ALGO, Auth::DEFAULT_ALGO_OPTIONS);
```

Comment:
It will make more sense to abstract all this logic in a validator. We should try to extend the current password validator and let the framework input validation take care of this or if not possible create a new validator and check inside the controller. Option 1 seems better if possible.

Reply:
updated.